packages/form-js-playground/src/Playground.js

```diff
@@ -34,12 +153,26 @@ function AppRoot(props) {
   const dataEditorRef = useRef();
   const resultViewRef = useRef();
 
-  const [ data, setData ] = useState(props.data || {});
+  const [ showEmbed, setShowEmbed ] = useState(false);
+
+  const [ initialData ] = useState(props.data || {});
+  const [ initialSchema, setInitialSchema ] = useState(props.schema);
 
+  const [ data, setData ] = useState(props.data || {});
   const [ schema, setSchema ] = useState(props.schema);
 
   const [ resultData, setResultData ] = useState(props.data || {});
 
+  useEffect(() => {
+    props.onInit({
+      setSchema: setInitialSchema
+    });
+  });
```

Comment:
Is this effect intended to run _on every_ render (because it's missing dependencies)? Setting the initial schema could maybe be a first render task.

Reply:
To my knowledge this is a once-only hook. But I'll investigate.